js/reference.js

```diff
@@ -2127,25 +2151,103 @@
 
                     // check url length limit if not first one;
                     if (i != 0 && new Reference(module.parse(baseUri + currentPath + (i != 0 ? ';' : '') + filter), self.table.schema.catalog).contextualize.compact._getReadPath().value.length > module.URL_PATH_LENGTH_LIMIT) {
-                        referencePaths.push(currentPath);
+                        // any more filters will go over the url length limit so save the current path and count
+                        // then clear both to start creating a new path
+                        referencePathObjs.push({
+                            path: baseUri + currentPath,
+                            count: currentCount
+                        });
+
                         currentPath = compactPath;
+                        currentCount = 0;
                     } else if (i != 0) {
                         // prepend the conjunction operator when it isn't the first filter to create and we aren't dealing with a url length limit
                         filter = ";" + filter;
                     }
 
+                    // append the filter either on the previous path after adding ";", or on the new path started from compactPath
                     currentPath += filter;
+                    currentCount++;
                 }
-                referencePaths.push(currentPath);
-
-                referencePaths.forEach(function (path) {
-                    var reference = new Reference(module.parse(baseUri + path), self.table.schema.catalog);
-                    references.push(reference);
+                // After last iteration of loop, push the current path
+                referencePathObjs.push({
+                    path: baseUri + currentPath,
+                    count: currentCount
                 });
 
-                return references;
+                var k = 0;
```

Comment:
This is not very important, but if you define all the variables in the beginning it makes the code a lot easier to read. I was struggling to find where you've instantiated `k` that is used in the `finally` clause in the `recursiveDelete` function.